vyper/contracts/libs/FulfillmentApplier.vy

```diff
@@ -0,0 +1,489 @@
+
+# Items that can be spent
+SIDE_OFFER: constant(uint8) = 0
+
+@internal
+@pure
+def _checkMatchingConsideration(
+    consideration: ReceivedItem,
+    receivedItem: ReceivedItem
+) -> bool:
+    """
+    @dev Internal pure function to check the indicated consideration item matches original item.
+
+    @param consideration  The consideration to compare
+    @param receivedItem  The aggregated received item
+
+    @return invalidFulfillment A boolean indicating whether the fulfillment is invalid.
+    """
+    return \
+        receivedItem.recipient != consideration.recipient or \
+        receivedItem.itemType != consideration.itemType or \
+        receivedItem.token != consideration.token or \
+        receivedItem.identifier != consideration.identifier
+
+
+@internal
+@view # convert to pure with next Vyper release
+def _aggregateValidFulfillmentConsiderationItems(
+    ordersToExecute: DynArray[OrderToExecute, 10],
+    considerationComponents: DynArray[FulfillmentComponent, 10],
+    startIndex: uint256
+) -> ReceivedItem:
+    """
+    @dev Internal pure function to aggregate a group of consideration items
+         using supplied directives on which component items are candidates
+         for aggregation, skipping items on orders that are not available.
+
+    @param ordersToExecute         The orders to aggregate consideration
+                                   items from.
+    @param considerationComponents An array of FulfillmentComponent structs
+                                   indicating the order index and item index
+                                   of each candidate consideration item for
+                                   aggregation.
+    @param startIndex              The initial order index to begin iteration
+                                   on when searching for consideration items
+                                   to aggregate.
+
+    @return receivedItem The aggregated consideration items.
+    """
+    # Declare struct into avoid declaring multiple local variables
+    potentialCandidate: ConsiderationItemIndicesAndValidity = \
+        ConsiderationItemIndicesAndValidity({
+            orderIndex: considerationComponents[startIndex].orderIndex,
+            itemIndex: considerationComponents[startIndex].itemIndex,
+            # Ensure that order index is in range.
+            invalidFulfillment: considerationComponents[startIndex].orderIndex >= len(ordersToExecute)
+        })
+
+    assert not potentialCandidate.invalidFulfillment, "invalid fulfillment component data"
+
+    receivedItem: ReceivedItem = empty(ReceivedItem)
+
+    # Retrieve relevant item using order index.
+    orderToExecute: OrderToExecute = ordersToExecute[potentialCandidate.orderIndex]
+    # Ensure that the item index is not out of range.
+    potentialCandidate.invalidFulfillment = \
+        potentialCandidate.invalidFulfillment or \
+        (potentialCandidate.itemIndex >= len(orderToExecute.receivedItems))
```

Comment:
I usually prefer using parenthesis over `\`
```suggestion
    potentialCandidate.invalidFulfillment = (
        potentialCandidate.invalidFulfillment
        or potentialCandidate.itemIndex >= len(orderToExecute.receivedItems)
    )
```

Reply:
updated. one case which I wasn't able to avoid as having the revert string in a different line in `assert`.